### ISSUE TITLE
tried dri and I get an error

### ISSUE THREAD

**Issue by montvid** [postmarketOS/osk-sdl#54]
I tried dri and I get an error. Can it be fixed?

*) Direct/Thread: Started 'SigHandler' (1384) [CRITICAL - OTHER/0] <85316>...

| DirectFB 1.7.7 |
(c) 2012-2015 DirectFB integrated media GmbH
(c) 2001-2015 The world wide DirectFB Open Source Community
(c) 2000-2004 Convergence (integrated media) GmbH
----------------------------------------------------------------

() DirectFB/Core: Single Application Core. (2017-10-03 12:19)
() Direct/Memcpy: Using armasm_memcpy()
() Direct/Thread: Started 'Fusion Dispatch' (1385) [MESSAGING - OTHER/0] <85316>...
() Direct/Thread: Started 'VT Switcher' (1389) [CRITICAL - OTHER/0] <85316>...
() Direct/Thread: Started 'VT Flusher' (1390) [DEFAULT - OTHER/0] <85316>...
() DRMKMS/Init: using device /dev/dri/card0 (default)
() Direct/Thread: Started 'DRMKMS/Buffer' (1391) [CRITICAL - OTHER/0] <85316>...
() Direct/Thread: Started 'tslib Input' (1392) [INPUT - OTHER/0] <85316>...
() DirectFB/Input: tslib touchscreen 0 0.1 (tslib)
() DirectFB/Input: Hot-plug detection enabled with Input Hub Driver
() DirectFB/Graphics: Generic Software Rasterizer 0.7 (directfb.org)
() DirectFB/DRMKMS: Got 1 connectors, 2 encoders
() DirectFB/DRMKMS: found connected connector id 32.
() DirectFB/DRMKMS: connector 32 is already bound to encoder 31.
() DirectFB/DRMKMS: encoder 31 is already bound to ctrc 28.
() DirectFB/DRMKMS: Default mode is 1200x1920, we have 1 modes in total
(!!!) *** ONCE [no mode found for 110x2560 at 60 Hz
] *** [drmkms_system.c:665 in drmkms_find_mode()]
() DirectFB/DRMKMS: supported properties for layer id 25
() DirectFB/DRMKMS: supported properties for layer id 26
() DirectFB/Core/WM: Default 0.3 (directfb.org)
() Direct/Thread: Started 'Genefx' (1393) [DEFAULT - OTHER/0] <85316>...
INFO: DirectFB Device Information
INFO: ===========================
INFO: Name: Software Rasterizer
INFO: Vendor: Generic
INFO: Driver Name: Software Driver
INFO: Driver Vendor: directfb.org
INFO: Driver Version: 0.7
INFO: Video memory: 0
INFO: Blitting flags:
INFO: Drawing flags:
INFO: Acceleration flags:
ERROR: SDL_GetDisplayMode failed: index must be in the range of 0 - -1

**Comment by craftyguy** [comment]
Looks like the problem is here: 

> (!!!) *** ONCE [no mode found for 110x2560 at 60 Hz

DirectFB (?) is returning a crazy mode for your device.

**Comment by montvid** [comment]
Well, yes. The question is how to detect the proper mode? I found this that could help but i don't know how to implement it:

https://tutel.me/c/programming/questions/37722003/sdl+desktop+resolution+detection+in+linux

**Comment by craftyguy** [comment]
That error is from DirectFB, not SDL, see this: https://github.com/DirectFB/directfb/blob/master/systems/drmkms/drmkms_system.c#L665

Just FYI, but this is likely going to be a pretty low priority for me at the moment. osk-sdl is meant to be only run on boot, and for only a short period of time. Some other work is going on now to reduce the CPU utilization when doing SW rendering and if that resolves your high CPU utilization then I don't think any immediate reason to pursue enabling/testing other directfb video systems (like drm/kms).

**Comment by montvid** [comment]
Yes, it is not high priority, just a reminder if you or someone stumbles upon a fix.

**Comment by craftyguy** [comment]
Oh yea, totally, thanks for reporting it 😄
